src/plugins/world_control/WorldControlEventListener.hh

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef IGNITION_GUI_WORLDCONTROLEVENTLISTENER_HH_
+#define IGNITION_GUI_WORLDCONTROLEVENTLISTENER_HH_
+
+#include "ignition/gui/Application.hh"
+#include "ignition/gui/Export.hh"
+#include "ignition/gui/GuiEvents.hh"
+#include "ignition/gui/MainWindow.hh"
+#include "ignition/gui/qt.h"
+
+namespace ignition
+{
+namespace gui
+{
+  /// \brief Helper class for testing listening to events emitted by the
+  /// WorldControl plugin. This is used for testing the event behavior of
+  /// the WorldControl plugin.
+  class IGNITION_GUI_VISIBLE WorldControlEventListener : public QObject
```

Comment:
Great work putting the tests together! Ideally we wouldn't include this as part of the plugin itself, since it's only used for testing. But I think this can be addressed later.

Reply:
Done in 8e7cb8f